pkg/services/ngalert/api/api_prometheus_test.go

```diff
@@ -411,9 +412,89 @@ func TestRouteGetRuleStatuses(t *testing.T) {
 `, folder.Fullpath), string(r.Body()))
 	})
 
+	t.Run("with a rule that is paused", func(t *testing.T) {
+		fakeStore, fakeAIM, api := setupAPI(t)
+		generateRuleAndInstanceWithQuery(t, orgID, fakeAIM, fakeStore, withClassicConditionSingleQuery(), gen.WithNoNotificationSettings(), gen.WithIsPaused(true))
+		folder := fakeStore.Folders[orgID][0]
+
+		r := api.RouteGetRuleStatuses(c)
+		require.Equal(t, http.StatusOK, r.Status())
+		require.JSONEq(t, fmt.Sprintf(`
+{
+	"status": "success",
+	"data": {
+		"groups": [{
+			"name": "rule-group",
+			"file": "%s",
+			"folderUid": "namespaceUID",
+			"rules": [{
+				"state": "inactive",
+				"name": "AlwaysFiring",
+				"folderUid": "namespaceUID",
+				"uid": "RuleUID",
+				"query": "vector(1)",
+				"queriedDatasourceUIDs": ["AUID"],
+				"alerts": [{
+					"labels": {
+						"job": "prometheus"
+					},
+					"annotations": {
+						"severity": "critical"
+					},
+					"state": "Normal",
+					"activeAt": "0001-01-01T00:00:00Z",
+					"value": ""
+				}],
+				"totals": {
+					"normal": 1
+				},
+				"totalsFiltered": {
+					"normal": 1
+				},
+				"labels": {
+					"__a_private_label_on_the_rule__": "a_value"
+				},
+				"health": "ok",
+				"isPaused": true,
+				"type": "alerting",
+				"lastEvaluation": "2022-03-10T14:01:00Z",
+				"duration": 180,
+				"keepFiringFor": 10,
+				"evaluationTime": 60
+			}],
+			"totals": {
+				"inactive": 1
+			},
+			"interval": 60,
+			"lastEvaluation": "2022-03-10T14:01:00Z",
+			"evaluationTime": 60
+		}],
+		"totals": {
+			"inactive": 1
+		}
+	}
+}
+`, folder.Fullpath), string(r.Body()))
+	})
+
+	t.Run("with a rule that has notification settings", func(t *testing.T) {
+		fakeStore, fakeAIM, api := setupAPI(t)
+		notificationSettings := ngmodels.NotificationSettings{
+			Receiver: "test-receiver",
+			GroupBy:  []string{"job"},
+		}
+		generateRuleAndInstanceWithQuery(t, orgID, fakeAIM, fakeStore, withClassicConditionSingleQuery(), gen.WithNotificationSettings(notificationSettings), gen.WithIsPaused(false))
+		r := api.RouteGetRuleStatuses(c)
+		require.Equal(t, http.StatusOK, r.Status())
+		var res apimodels.RuleResponse
+		require.NoError(t, json.Unmarshal(r.Body(), &res))
+		require.NotNil(t, res.Data.RuleGroups[0].Rules[0].NotificationSettings)
```

Comment:
Nit:
Do a quick check on `len(res.Data.RuleGroups)` and `len(res.Data.RuleGroups[0].Rules` just so there are no panics in case something goes wrong.